src/datatype.ts

```diff
@@ -0,0 +1,260 @@
+import type { Faker } from '.';
+
+export class Datatype {
+  constructor(private readonly faker: Faker, seed?: any[] | any) {
+    // Use a user provided seed if it is an array or number
+    if (Array.isArray(seed) && seed.length) {
+      this.faker.mersenne.seed_array(seed);
+    } else if (!isNaN(seed)) {
+      this.faker.mersenne.seed(seed);
+    }
+
+    // Bind `this` so namespaced is working correctly
+    for (const name of Object.getOwnPropertyNames(Datatype.prototype)) {
+      if (name === 'constructor' || typeof this[name] !== 'function') {
+        continue;
+      }
+      this[name] = this[name].bind(this);
+    }
+  }
+
+  /**
+   * Returns a single random number based on a max number or range.
+   *
+   * @method faker.datatype.number
+   * @param options
+   */
+  number(
+    options?: number | { min?: number; max?: number; precision?: number }
+  ): number {
+    if (typeof options === 'number') {
+      options = { max: options };
+    }
+
+    options = options ?? {};
+
+    if (typeof options.min === 'undefined') {
+      options.min = 0;
+    }
+
+    if (typeof options.max === 'undefined') {
+      options.max = 99999;
+    }
+
+    if (typeof options.precision === 'undefined') {
+      options.precision = 1;
+    }
+
+    // Make the range inclusive of the max value
+    let max = options.max;
+    if (max >= 0) {
+      max += options.precision;
+    }
+
+    let randomNumber = Math.floor(
+      this.faker.mersenne.rand(
+        max / options.precision,
+        options.min / options.precision
+      )
+    );
+    // Workaround problem in Float point arithmetics for e.g. 6681493 / 0.01
+    randomNumber = randomNumber / (1 / options.precision);
+
+    return randomNumber;
+  }
+
+  /**
+   * Returns a single random floating-point number based on a max number or range.
+   *
+   * @method faker.datatype.float
+   * @param options
+   */
+  float(
+    options?: number | { min?: number; max?: number; precision?: number }
+  ): number {
+    if (typeof options === 'number') {
+      options = {
+        precision: options,
+      };
+    }
+    options = options || {};
+    const opts: { precision?: number } = {};
+    for (const p in options) {
+      opts[p] = options[p];
+    }
+    if (typeof opts.precision === 'undefined') {
+      opts.precision = 0.01;
+    }
+    return this.faker.datatype.number(opts);
+  }
+
+  /**
+   * Returns a Date object using a random number of milliseconds since 1. Jan 1970 UTC
+   * Caveat: seeding is not working
+   *
+   * @method faker.datatype.datetime
+   * @param options pass min OR max as number of milliseconds since 1. Jan 1970 UTC
+   */
+  datetime(options): Date {
+    if (typeof options === 'number') {
+      options = {
+        max: options,
+      };
+    }
+
+    const minMax = 8640000000000000;
```

Comment:
```suggestion
    const minMax = Number.MAX_SAFE_INTEGER;
```

equal to 9007199254740991